txscript/opcode.go

```diff
@@ -2045,12 +2045,30 @@ func opcodeCheckSig(op *parsedOpcode, vm *Engine) error {
 	// Get script starting from the most recent OP_CODESEPARATOR.
 	subScript := vm.subScript()
 
-	// Remove the signature since there is no way for a signature to sign
-	// itself.
-	subScript = removeOpcodeByData(subScript, fullSigBytes)
+	// With BIP 143, Bitcoin Core's 'FindAndDelete' function is no longer
```

Comment:
I don't think we should call out the function name in another code base which would force the reader to go look somewhere else.  Also, it could easily change at some point further reducing its effectiveness.  I think this should instead just read something like:

```Go
	// Remove the signature in pre-segwit scripts since there is no way for
	// a signature to sign itself.
	if !vm.witness {
		subScript = removeOpcodeByData(subScript, fullSigBytes)
	}
```